src/Symfony/Component/VarExporter/Instantiator.php

```
@@ -0,0 +1,104 @@
+     *
+     * @param array $propertiesByClass The properties to set on the object, keyed by the class to bind to
+     *                                 when setting them. The first key of the array defines the class
+     *                                 of the returned instance.
+     *
+     * @return object An instance of the class defined by the first key of $propertiesByClass
+     *
+     * @throws \ReflectionException When the instance cannot be created
+     */
+    public static function instantiate(array $propertiesByClass)
+    {
+        if (!$propertiesByClass) {
+            throw new \ReflectionException('No class provided.');
+        }
+
+        foreach ($propertiesByClass as $class => $properties) {
+            break;
+        }
+
+        try {
+            $reflector = Registry::$reflectors[$class] ?? Registry::getClassReflector($class);
+        } catch (\Exception $e) {
+            if (0 === strpos($e->getMessage(), 'Serialization of ')) {
+                $e = new \ReflectionException(substr_replace($e->getMessage(), 'Instanti', 0, 8));
+            }
+
+            throw $e;
+        }
+
+        if (Registry::$cloneable[$class]) {
+            $instance = clone Registry::$prototypes[$class];
+        } elseif (Registry::$instantiableWithoutConstructor[$class]) {
+            $instance = $reflector->newInstanceWithoutConstructor();
+        } elseif (null === Registry::$prototypes[$class]) {
+            throw new \ReflectionException(sprintf("Instantiation of '%s' is not allowed", $class));
+        } elseif ($reflector->implementsInterface('Serializable')) {
+            $instance = unserialize('C:'.\strlen($class).':"'.$class.'":0:{}');
+        } else {
+            $instance = unserialize('O:'.\strlen($class).':"'.$class.'":0:{}');
+        }
+
+        $instance = array($instance);
+
+        foreach ($propertiesByClass as $class => $properties) {
+            if (!$properties) {
```

Would probably rather throw an exception: why is a key even declared if nothing is to be hydrated for this inheritance level?

---

There would be no mistake so an exception would be too strict here. Think of generated values: forcing a non-empty array here would increase the burden on the caller to prepare its value, for no logical (no "mathematical") reason. An empty array is valid input to me.